src/app/tools/point_shapes.h

```diff
@@ -100,18 +100,32 @@ class BrushPointShape : public PointShape {
         const float ti = 1.0f - pt.gradient;
         switch (loop->sprite()->pixelFormat()) {
           case IMAGE_RGB:
-            if (rgba_geta(a) == 0) a = b;
-            else if (rgba_geta(b) == 0) b = a;
-            a = doc::rgba(int(ti*rgba_getr(a) + t*rgba_getr(b)),
-                          int(ti*rgba_getg(a) + t*rgba_getg(b)),
-                          int(ti*rgba_getb(a) + t*rgba_getb(b)),
-                          int(ti*rgba_geta(a) + t*rgba_geta(b)));
+            if (rgba_geta(a) == 0)
+              a = doc::rgba(rgba_getr(b),
+                            rgba_getg(b),
+                            rgba_getb(b),
+                            int(t*rgba_geta(b)));
+            else if (rgba_geta(b) == 0)
+              a = doc::rgba(rgba_getr(a),
+                            rgba_getg(a),
+                            rgba_getb(a),
+                            int(ti*rgba_geta(a)));
+            else
```

Comment:
Why is this needed? I tried this PR and then tried again without these lines...I couldn't say the difference...

Reply:
If you comment these lines, you'll get a mix between a transparent black and some opaque color (I use Alpha Compositing Ink)
<img width="175" alt="Screen Shot 2022-09-19 at 14 11 02" src="https://user-images.githubusercontent.com/32165475/191074422-daeb4e28-b9db-4a29-b0c7-c31c6c901af2.png">

With this new lines you'll get a mix between the desired opaque color and the same but full transparent (I use Alpha Compositing Ink):
<img width="175" alt="Screen Shot 2022-09-19 at 14 18 39" src="https://user-images.githubusercontent.com/32165475/191075823-6471d8eb-5160-4777-a698-4a94f5beca58.png">

Anyway, by explaining this, I now understand what David intended with the original code. I'll try to simplify the code and then force a new commit.

Reply:
I see now, thanks!

Reply:
I found
```
if (rgba_geta(a) == 0)
  a = doc::rgba(rgba_getr(b),
                rgba_getg(b),
                rgba_getb(b),
                int(t*rgba_geta(b)));
```
it's fine, because another more general approach will require more math to cancel members.